### DISCUSSION TITLE
SSR wihtout Next.js with multi-pass render util

### DISCUSSION THREAD

**Discussion by shiro** [trpc/trpc#2478] [General]
Hi, thanks for the amazing work on the lib!

I'm currently using apollo with `getMarkupFromTree`, which wraps React's `renderToString`, injects an SSR react context to store promises, then proceeds to multi-pass render, stopping at queries and continuing when promises resolve.

I wanted to try out tRPC, but switching to Next.js is not feasible right now and I need SSR without having to manually type out the queries (obviously not doable for a big project). Is there any way to do this/are there plains to add this?

**Comment by KATT** [comment]
We follow the same principle with our SSR (link below), would probably not be too hard to make this a standalone thing that could be used as you describe here, but will probably not come from us in the core team. There are so many near-future things in React with RSC etc that might make that work be in vain. 

https://trpc.io/docs/ssr
